### PR TITLE
Document the side effect of client.getMetadata (#68)

### PR DESCRIPTION
**Pull request by edoardocomar**
Addresses issue #68 

investigated with @mimaison

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -304,6 +304,14 @@ Client.prototype.disconnect = function(cb) {
 
 /**
  * Get client metadata.
+ * <p/>
+ * <b>Note:</b> using a <code>metadataOptions.topic</code> parameter has a potential side-effect.
+ * A Topic object will be created, if it did not exists yet, with default options
```
Comment (webmakersteve) [inline lib/client.js]: Typo: A topic object will be created, if it did not *exist* yet.

File: README.md [hunk 2]
```diff
@@ -406,6 +406,8 @@ Getting metadata on any connection returns the following data structure:
 
 The following example illustrates how to use the `getMetadata` method.
 
+Please see the documentation for `Client.getMetadata` for a potential side-effect of the call.
```
Comment (webmakersteve) [inline README.md]: Can we reword this to something a little clearer. This seems a bit scary:

```
If you intend on passing a topic configuration to a topic you are explicitly fetching metadata for, be sure to created that topic prior to calling getMetadata on it. When fetching metadata for a topic, if a topic reference does not exist, one is created using the default config, which makes subsequent topic calls simply return that object.
```

We can and should also document it in the docblock as you have, but I think it's less cryptic if it is explicit in the README as well.

File: lib/client.js (hunk 1 above)
Comment (webmakersteve) [inline lib/client.js]: I have mixed feelings about using HTML markup in this comment when it isn't used in other places. At the very least, can we remove the `<p/>` and just leave the comment formatted as the other ones are?

**Comment by webmakersteve** [comment]
Some comments on the PR and then I will be happy to merge.

File: lib/client.js (hunk 1 above)
Comment (edoardocomar) [inline lib/client.js]: ok, np!

File: README.md (hunk 2 above)
Comment (edoardocomar) [inline README.md]: ok, made it a bit more explicit, less scary ;-) but still pointing to the docs. Duplication is evil ...

**Comment by edoardocomar** [comment]
@webmakersteve changes ok?
